File: solutions/vbnet/secret-handshake/1/SecretHandshake.vb
Imports System
Imports System.Collections.Generic

Public Module SecretHandshake
    Private ReadOnly orders As IDictionary(Of Integer, String) = New Dictionary(Of Integer, String) From {
        {&B1, "wink"},
        {&B10, "double blink"},
        {&B100, "close your eyes"},
        {&B1000, "jump"}
    }

    Public Function Commands(ByVal commandValue As Integer) As String()
        Dim s = New List(Of String)()

        For Each order In orders
            If (commandValue And order.Key) = order.Key Then s.Add(order.Value)
        Next

        If (commandValue And &B10000) = &B10000 Then s.Reverse()
        Return s.ToArray()
    End Function
End Module

File: solutions/euphoria/secret-handshake/1/secret-handshake.ex
include std/math.e
include std/sequence.e

sequence operations = {"wink", "double blink", "close your eyes", "jump"}

public function commands(integer number)
  sequence res = {}
  for i = 1 to 4 do
    if and_bits(number, power(2, i - 1)) > 0 then
      res &= {operations[i]}
    end if
  end for
  if and_bits(number, 16) > 0 then
    res = reverse(res)
  end if
  return res
end function